src/manage.c

```diff
@@ -2977,26 +2977,46 @@ slave_setup (gvm_connection_t *connection, const char *name, task_t task,
               privacy_algorithm
                 = credential_iterator_privacy_algorithm (&credentials);
 
-              if (community == NULL || user == NULL || password == NULL
-                  || auth_algorithm == NULL || privacy_password == NULL
-                  || privacy_algorithm == NULL)
+              if (password && strcmp (password, "")
+                  && (auth_algorithm == NULL || strcmp (auth_algorithm, "")))
                 {
                   cleanup_iterator (&credentials);
                   global_slave_esxi_credential_uuid = NULL;
                   g_warning ("Could not create slave SNMP credential"
-                             " (missing community, login, password,"
-                             " auth_algorithm, privacy_password"
-                             " or privacy_algorithm)."
+                             " (auth_algorithm must be set if password is"
+                             " given)."
+                             " Continuing without credential.");
+                }
+              if (((privacy_password && strcmp (privacy_password, ""))
+                    || (privacy_algorithm && strcmp (privacy_algorithm, "")))
+                  && (password == NULL
+                      || auth_algorithm == NULL
+                      || privacy_password == NULL
+                      || privacy_algorithm == NULL))
+                {
+                  cleanup_iterator (&credentials);
+                  slave_esxi_credential_uuid = NULL;
```

Comment:
Also should be snmp.

Reply:
I've changed this and the above in d01d4f3.